boards/arm/nucleo_h745zi_q/doc/index.rst

```diff
@@ -0,0 +1,254 @@
+Nucleo H745ZI-Q System Clock could be driven by an internal or external
+oscillator, as well as the main PLL clock. By default, the System clock is
+driven by the PLL clock at 96MHz, driven by an 8MHz high-speed external clock.
+
+Serial Port
+-----------
+
+Nucleo H745ZI-Q board has 4 UARTs and 4 USARTs. The Zephyr console output is
+assigned to UART3. Default settings are 115200 8N1.
+
+Resources sharing
+-----------------
+
+The dual core nature of STM32H745 SoC requires sharing HW resources between the
+two cores. This is done in 3 ways:
+
+- **Compilation**: Clock configuration is only accessible to M7 core. M4 core only
+  has access to bus clock activation and deactivation.
+- **Static pre-compilation assignment**: Peripherals such as a UART are assigned in
+  devicetree before compilation. The user must ensure peripherals are not assigned
+  to both cores at the same time.
+- **Run time protection**: Interrupt-controller and GPIO configurations could be
+  accessed by both cores at run time. Accesses are protected by a hardware semaphore
+  to avoid potential concurrent access issues.
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_h745zi_q`` board should be built per core target,
+using either ``nucleo_h745zi_q_m7`` or ```nucleo_h745zi_q_m4`` as the target
+(see :ref:`build_an_application` and :ref:`application_run` for more details).
+
+.. note::
+
+   If using OpenOCD you will need a recent development version as the last
+   official release does not support H7 series and ST-LINK V3 yet.
+   Following links may be helpful: `OpenOCD installing Devug Version`_
+                               and `OpenOCD installing with ST-LINK V3 support`_
+
+Flashing
+========
+
+Nucleo H745ZI-Q board includes an ST-LINK/V3 embedded debug tool interface.
+
+Flashing operation will depend on the target to be flashed and the SoC
+option bytes configuration.
+
+By default:
+
+  - CPU1 (Cortex-M7) boot address is set to 0x80000000 (OB: BOOT_CM7_ADD0)
+  - CPU2 (Cortex-M4) boot address is set to 0x81000000 (OB: BOOT_CM4_ADD0)
+
+Also, default out of the box board configuration enables CM7 and CM4 boot when
+board is powered (Option bytes BCM7 and BCM4 are checked).
+In that configuration, Kconfig boot option ``STM32H7_BOOT_CM4_CM7`` should be selected.
+Zephyr flash configuration has been set to meet these default settings.
+
+Flashing an application to STM32H745ZI M7 Core
+----------------------------------------------
+First, connect the NUCLEO-H745ZI-Q to your host computer using
+the USB port to prepare it for flashing. Then build and flash your application.
+
+Here is an example for the :ref:`hello_world` application.
+
+Run a serial host program to connect with your NUCLEO-H745ZI-Q board.
+
+.. code-block:: console
+
+   $ minicom -b 115200 -D /dev/ttyACM0
+
+Build and flash the application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_h745zi_q_m7
+   :goals: build flash
```

Comment:
If I'm correct, you didn't provide any runner configuration (board.cmake), so I wonder if west flash actually works as described

Reply:
At least my blinking-diode example are working on both M7 and M4. Before every flashing operation I did erase of flash.

Reply:
added runner configuration